packages/flow/lib/types.ts

```diff
@@ -0,0 +1,23 @@
+type ScalarValues = string | number | bigint | boolean | symbol | null;
+
+interface HasForEachMethod {
+	forEach(callbackfn: (...args: Array<unknown>) => void, thisArg?: unknown): void;
+}
```

Comment:
This is really neat idea. What made you want to switch away from the built-in `Iterable` to this? Was it to support a looser range of values?

Reply:
The problem is that you have to create an overload or a union type. This solution forces the developer to use a data structure. The best solution to this problem would have been to find the typing for the `for of` loop. But i couldn't find it. 
This solution makes it so that the type for data structure is inferred.